File: Controllers/AdminController.cs
using System.Web.Mvc;
using MyWeb.Models;

namespace MyWeb.Controllers
{
    [Authorize(Roles = "Admin")]
    public class AdminController : Controller
    {
        // GET: /Admin
        public ActionResult Index()
        {
            return View();
        }

        // --- 商品管理 ---

        // GET: /Admin/Products
        public ActionResult Products(int? page)
        {
            // TODO: 查詢所有商品
            return View();
        }

        // GET: /Admin/ProductEdit/5
        public ActionResult ProductEdit(int? id)
        {
            // TODO: 查詢商品（id=null 為新增）
            return View();
        }

        // POST: /Admin/ProductEdit
        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult ProductEdit(Product model)
        {
            if (!ModelState.IsValid) return View(model);
            // TODO: 儲存商品資料
            return RedirectToAction("Products");
        }

        // --- 訂單管理 ---

        // GET: /Admin/Orders
        public ActionResult Orders(string status, int? page)
        {
            // TODO: 查詢所有訂單
            return View();
        }

        // GET: /Admin/OrderDetail/5
        public ActionResult OrderDetail(int id)
        {
            // TODO: 查詢訂單詳細
            return View();
        }

        // POST: /Admin/UpdateOrderStatus
        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult UpdateOrderStatus(int id, string status)
        {
            // TODO: 更新訂單狀態
            return RedirectToAction("OrderDetail", new { id = id });
        }

        // --- 會員管理 ---

        // GET: /Admin/Members
        public ActionResult Members(int? page)
        {
            // TODO: 查詢所有會員
            return View();
        }

        // --- 部落格管理 ---

        // GET: /Admin/Blogs
        public ActionResult Blogs(int? page)
        {
            return View();
        }

        // GET: /Admin/BlogEdit/5
        public ActionResult BlogEdit(int? id)
        {
            return View();
        }

        // POST: /Admin/BlogEdit
        [HttpPost]
        [ValidateAntiForgeryToken]
        public ActionResult BlogEdit(BlogPost model)
        {
            if (!ModelState.IsValid) return View(model);
            // TODO: 儲存文章
            return RedirectToAction("Blogs");
        }
    }
}

File: Controllers/AdminController.vb
Imports System.Web.Mvc
Imports My.Models

Namespace My.Controllers
    <Authorize(Roles:="Admin")>
    Public Class AdminController
        Inherits Controller

        ' GET: /Admin
        Function Index() As ActionResult
            Return View()
        End Function

        ' --- 商品管理 ---

        ' GET: /Admin/Products
        Function Products(page As Integer?) As ActionResult
            ' TODO: 查詢所有商品
            Return View()
        End Function

        ' GET: /Admin/ProductEdit/5
        Function ProductEdit(id As Integer?) As ActionResult
            ' TODO: 查詢商品（id=Nothing 為新增）
            Return View()
        End Function

        ' POST: /Admin/ProductEdit
        <HttpPost>
        <ValidateAntiForgeryToken>
        Function ProductEdit(model As Product) As ActionResult
            If Not ModelState.IsValid Then Return View(model)
            ' TODO: 儲存商品資料
            Return RedirectToAction("Products")
        End Function

        ' --- 訂單管理 ---

        ' GET: /Admin/Orders
        Function Orders(status As String, page As Integer?) As ActionResult
            ' TODO: 查詢所有訂單
            Return View()
        End Function

        ' GET: /Admin/OrderDetail/5
        Function OrderDetail(id As Integer) As ActionResult
            ' TODO: 查詢訂單詳細
            Return View()
        End Function

        ' POST: /Admin/UpdateOrderStatus
        <HttpPost>
        <ValidateAntiForgeryToken>
        Function UpdateOrderStatus(id As Integer, status As String) As ActionResult
            ' TODO: 更新訂單狀態
            Return RedirectToAction("OrderDetail", New With {.id = id})
        End Function

        ' --- 會員管理 ---

        ' GET: /Admin/Members
        Function Members(page As Integer?) As ActionResult
            ' TODO: 查詢所有會員
            Return View()
        End Function

        ' --- 部落格管理 ---

        ' GET: /Admin/Blogs
        Function Blogs(page As Integer?) As ActionResult
            Return View()
        End Function

        ' GET: /Admin/BlogEdit/5
        Function BlogEdit(id As Integer?) As ActionResult
            Return View()
        End Function

        ' POST: /Admin/BlogEdit
        <HttpPost>
        <ValidateAntiForgeryToken>
        Function BlogEdit(model As BlogPost) As ActionResult
            If Not ModelState.IsValid Then Return View(model)
            ' TODO: 儲存文章
            Return RedirectToAction("Blogs")
        End Function

    End Class
End Namespace
